libs/langchain-google-common/src/connection.ts

```diff
@@ -277,6 +281,57 @@ export abstract class AbstractGoogleLLMConnection<
     return parameters.safetySettings ?? [];
   }
 
+  // Borrowed from the OpenAI invocation params test
+  isStructuredToolArray(tools?: unknown[]): tools is StructuredToolInterface[] {
+    return (
+      tools !== undefined &&
+      tools.every((tool) =>
+        Array.isArray((tool as StructuredToolInterface).lc_namespace)
+      )
+    );
+  }
+
+  structuredToolToFunctionDeclaration(
+    tool: StructuredToolInterface
+  ): GeminiFunctionDeclaration {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const jsonSchema = zodToJsonSchema(tool.schema) as any;
```

Comment:
Why is this being casted as `any`? The built in types should be enough?

Reply:
The issue is that I needed to remove two attributes from `jsonSchema` (which happens on line 301). Leaving it in as the built-in type was giving me errors in typescript (or in my IDE's linting of it).